DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js

```diff
@@ -10,4 +10,143 @@
+            }            
+        });
+    }
+}
+
+// Creates pagination html for the user to see other charts by taking the existing number of pages
+// and the active page the user is on.
+function makePagination(numberOfPages,activeNumber) {
+    const pagination = document.getElementById("pagination");
+    var paginationString = "";
+    paginationString += "<input type=\"button\" id=\"previous\" onclick=\"previousPage()\" value=\"previous\" />";
+    paginationString += "<a id=\"activeNumber\">  " + (activeNumber + 1)  +"  </a>"; 
+    paginationString += "<input type=\"button\" id=\"next\" onclick=\"nextPage(" + numberOfPages + ")\" value=\"next\" />";
+    pagination.innerHTML = paginationString;
+}
+
+// Increases the page forward by increasing the active page by one and letting the display
+// charts increase to the next two and rerunning getDSACampaigns to retrieve the charts.
+function nextPage(numberOfPages) {
+    const activePage = document.getElementById("activeNumber");
+    var increasePage = parseInt(activePage.innerText);
+    
+    // The if statement checks if the user clicks nextpage but there is
+    // no more charts to look at using numberOfPages.
+    if(increasePage + 1 <= numberOfPages){
+        increasePage++;  
+        secondChart += 2;
+        firstChart += 2;  
+    }
+    activePage.innerText = increasePage;
+    getDSACampaigns();
+}
+
+// Decreases the page backward by decreasing the active page by one and letting the display
+// charts decrease to the previous two and rerunning getDSACampaigns to retrieve the charts.
+function previousPage() {
+    const activePage = document.getElementById("activeNumber");
+    var decreasePage = parseInt(activePage.innerText);
+
+    // The if statement checks if the user clicks previousPage but there is
+    // no more charts to look at using 0 because there are no negative pages.
+    if(decreasePage - 1 > 0){
+        decreasePage--;  
+        secondChart -= 2;
+        firstChart -= 2;  
+    }
+    activePage.innerText = decreasePage;
+    getDSACampaigns();
+}
+
+function drawNewBarGraph(DSACampaign, chartNumber) {
+
+    var data = new google.visualization.DataTable();
+    data.addColumn('string', 'DSA Campaign');
+    data.addColumn('number', 'Impressions');
+    data.addColumn('number', 'Clicks');
+    data.addColumn('number', 'Cost (USD)');
+
+    data.addRow([DSACampaign.name, DSACampaign.impressions, DSACampaign.clicks, DSACampaign.cost]);
+
+    var options = {
+        chart: {
+            title: 'Statistics',
+            subtitle: 'Impressions, Clicks, and Cost',
+        },
+        bars: 'horizontal' // Required for Material Bar Charts.
+    };
+
+    var chart = new google.charts.Bar(document.getElementById('bar-chart' + chartNumber));
+    chart.draw(data, google.charts.Bar.convertOptions(options));
+    console.log("Drew bar graph.");
+
+    drawNewTable(DSACampaign, chartNumber);
```

Comment:
Remove from this function and have as a standalone.

Reply:
I would leave it here since the graph and table should be made and displayed at the same time.

Reply:
In that case can we have a overarching function that makes it clear that we're generating a bar graph and the table. That way it's clear what is happening from the function name, as generating the table as a side-effect of "drawNewBarGraph" is not clear.